src/test/fuzz/p2p_transport_deserializer.cpp

```diff
@@ -31,16 +31,19 @@ void test_one_input(const std::vector<uint8_t>& buffer)
         n_bytes -= handled;
         if (deserializer.Complete()) {
             const int64_t m_time = std::numeric_limits<int64_t>::max();
-            const CNetMessage msg = deserializer.GetMessage(Params().MessageStart(), m_time);
-            assert(msg.m_command.size() <= CMessageHeader::COMMAND_SIZE);
-            assert(msg.m_raw_message_size <= buffer.size());
-            assert(msg.m_raw_message_size == CMessageHeader::HEADER_SIZE + msg.m_message_size);
-            assert(msg.m_time == m_time);
-            if (msg.m_valid_header) {
-                assert(msg.m_valid_netmagic);
-            }
-            if (!msg.m_valid_netmagic) {
-                assert(!msg.m_valid_header);
+            auto result = deserializer.GetMessage(Params().MessageStart(), m_time);
+            if (result) {
+                const CNetMessage msg = *result;
```

Comment:
I've changed this so that it uses list initialization instead and then reaches inside the optional.  Now it's consistent with the "real" code in net.cpp.